pages/learn/deploy/offline-updates.md

```diff
@@ -0,0 +1,269 @@
+$ balena util available-drives
+/dev/...
+
+# Please make sure `--drive` is really the disk you want to write to. 
+$ sudo balena local flash ${tmpimg} \
+    --drive /dev/{{flash-usb-disk}}
+
+$ rm ${tmpimg}
+```
+
+### Process of reprovisioning 
+
+> Warning: For devices with internal storage this procedure erases the internal media of your device. Hence, remove any mass storage devices containing user data.
+
+With the update media ready having the latest release of the application preloaded. Follow the device's provisioning instructions as present on balenaCloud dashboard. 
+
+For example: For Raspberrypi devices, insert the recently flashed SD card and power up the device.
+
+### Update Device Registration(s)
+
+This section deals with device registration.
+
+>  manually reflect the new state of the device in balenaCloud for offline devices or devices on private networks without Internet access
+
+#### [Optional] patch state endpoint
+
+> strictly not required for devices with low bandwidth Internet connections as they will update the state endpoint automatically
+
+```bash
+$ os_version_tag=$(echo ${os_version} | awk -F'+' '{print $1}')
+$ os_version_semver=$(echo ${os_version_tag} | sed 's/v//g')
+$ os_revision=$(echo ${os_version} | awk -F'+' '{print $2}' | awk -F'.' '{print $1}')
+$ os_variant=$(echo ${os_version} | awk -F'+' '{print $2}' | awk -F'.' '{print $2}')
+
+$ supervisor_version=$(curl --silent https://raw.githubusercontent.com/balena-os/meta-balena/${os_version_tag}/meta-balena-common/recipes-containers/resin-supervisor/resin-supervisor.inc \
+    | grep SUPERVISOR_TAG \
+    | awk '{print $3}' \
+    | sed 's/"//g' \
+    | sed 's/v//g')
+
+  # (e.g.) staging
+$ release_id=$(curl --silent \
+	  "https://api.balena-staging.com/v6/release?\$filter=startswith(commit,'${commit}')&\$select=id" \
+	  -H "Authorization: Bearer $(cat ~/.balena/token)" | jq -r .d[].id)
+
+$ curl --silent \
+	  -X 'PATCH' "https://api.balena-staging.com/v6/device(uuid='${uuid}')" \
+	  -H "Authorization: Bearer $(cat ~/.balena/token)" \
+	  -H 'Content-type: application/json' \
+	  --data-binary "{\"os_variant\":\"${os_variant}\",\"os_version\":\"balenaOS ${os_version_semver}+${os_revision}\",\"supervisor_version\":\"${supervisor_version}\",\"is_running__release\":${release_id}}"
+```
+
+#### [Optional] set tags
+
+```bash
+$ balena tag set 'offline:commit' "${commit}" \
+  --device ${uuid}
+
+$ balena tag set 'offline:hostOS' "${os_version}" \
+  --device ${uuid}
+```
+
+Read more about it: blog link goes here
+Here's an example project to test things out https://github.com/balena-io-playground/offline-updates/
+
+
+
+# Reflash any device offline without losing its apps or identity
+Today, when you reflash a previously deployed device, you’re resetting it to a factory default state. Your device gets a new identity, a new API key, and updated `config.json` settings. All your apps, data and logs stored on the device are lost. 
```

Comment:
I would remove the statement since this doesn't effect anything user side tbh. Their device work need the key to talk to the API as long as that works, I think we should be good

Reply:
@ab77 Check it out now, how's this for wording I have moved this section to the top as the overview of the entire process. (2nd para)